api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java

```diff
@@ -65,4 +70,129 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
 		}
 		
 	}
+	
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test(expected = APIException.class)
+	public void getOrderGroupsByEncounter_shouldFailGivenNullEncounter() {
+		dao.getOrderGroupsByEncounter(null);
+	}
+	
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test(expected = APIException.class)
+	public void getOrderGroupsByPatient_shouldFailGivenNullPatient() {
+		dao.getOrderGroupsByPatient(null);
+	}
+	
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldGetOrderGroupsFromAnEncounter() {
+		OrderGroup orderGroupA = new OrderGroup();
+		OrderGroup orderGroupB = new OrderGroup();
+		OrderGroup orderGroupC = new OrderGroup();
+		
+		Encounter existingEncounter1 = Context.getEncounterService().getEncounter(3);
+		
+		orderGroupA.setEncounter(existingEncounter1);
+		orderGroupB.setEncounter(existingEncounter1);
+		orderGroupC.setEncounter(existingEncounter1);
+		
+		assertEquals(0, dao.getOrderGroupsByEncounter(existingEncounter1).size());
+		
+		dao.saveOrderGroup(orderGroupA);
+		dao.saveOrderGroup(orderGroupB);
+		dao.saveOrderGroup(orderGroupC);
+		
+		assertEquals(3, dao.getOrderGroupsByEncounter(existingEncounter1).size());
+	}
+	
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldGetOrderGroupsFrommMultipleEncounters() {
+		OrderGroup orderGroupA = new OrderGroup();
+		OrderGroup orderGroupB = new OrderGroup();
+		OrderGroup orderGroupC = new OrderGroup();
+		
+		Encounter existingEncounter1 = Context.getEncounterService().getEncounter(3);
+		Encounter existingEncounter2 = Context.getEncounterService().getEncounter(4);
+		
+		orderGroupA.setEncounter(existingEncounter1);
+		orderGroupB.setEncounter(existingEncounter2);
+		orderGroupC.setEncounter(existingEncounter1);
+		
+		assertEquals(0, dao.getOrderGroupsByEncounter(existingEncounter1).size());
+		assertEquals(0, dao.getOrderGroupsByEncounter(existingEncounter2).size());
+		
+		dao.saveOrderGroup(orderGroupA);
+		dao.saveOrderGroup(orderGroupB);
+		dao.saveOrderGroup(orderGroupC);
+		
+		assertEquals(2, dao.getOrderGroupsByEncounter(existingEncounter1).size());
+		assertEquals(1, dao.getOrderGroupsByEncounter(existingEncounter2).size());
+	}
+	
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByPatient_shouldGetOrderGroupsFromMultiplePatients() {
+		OrderGroup orderGroupA = new OrderGroup();
```

Comment:
Instead of creating and saving this on the fly before fetching them, why don't you just out them in a dataset file and fetch them from there? Here are examples of such files: https://github.com/openmrs/openmrs-core/tree/master/api/src/test/resources/org/openmrs/api/include

Reply:
@dkayiwa i have added dataset, requesting for your review